server_addon/maya/client/ayon_maya/plugins/publish/collect_yeti_rig.py

```diff
@@ -304,3 +310,84 @@ def _replace_tokens(self, strings):
                     raise RuntimeError(msg)
             replaced.append(s)
         return replaced
+
+
+class CollectOxRig(CollectYetiRig):
+    """Collect all information of the Ornatrix Rig"""
+
+    order = pyblish.api.CollectorOrder + 0.4
+    label = "Collect Ornatrix Rig"
+    families = ["OxRig"]
+
+    def process(self, instance):
+        assert "input_SET" in instance.data["setMembers"], (
+            "Ornatrix Rig must have an input_SET")
+
+        ornatrix_nodes = cmds.ls(instance[:], long=True)
+        self.log.debug(f"Getting ornatrix nodes: {ornatrix_nodes}")
+        # Force frame range for yeti cache export for the rig
+        # Collect any textures if used
+        ornatrix_resources = []
+        for node in ornatrix_nodes:
+            # Get Yeti resources (textures)
+            resources = self.get_texture_resources(node)
+            ornatrix_resources.extend(resources)
+        # avoid duplicate dictionary data
+        instance.data["resources"] = [
+            i for n, i in enumerate(ornatrix_resources)
+            if i not in ornatrix_resources[n + 1:]
+        ]
+        self.log.debug("{}".format(instance.data["resources"]))
+        start = cmds.playbackOptions(query=True, animationStartTime=True)
+        for key in ["frameStart", "frameEnd",
+                    "frameStartHandle", "frameEndHandle"]:
+            instance.data[key] = start
+
+    def get_texture_resources(self, node):
+        # add docstrings
+        resources = []
+        node_shape = cmds.listRelatives(node, shapes=True)
+        if not node_shape:
+            return []
+
+        ox_nodes = [
+            ox_node for ox_node in cmds.listConnections(node_shape, destination=True)
+                    if cmds.nodeType(ox_node) in ORNATRIX_NODES
+        ]
+        ox_imageFile = [
+            ox_img for ox_img in cmds.listConnections(ox_nodes, destination=False)
+            if cmds.nodeType(ox_img) == "file"
+        ]
```

Comment:
Instead of checking nodetype one by one you're better of just doing a `cmds.ls(type=ORNATRIX_NODES)` query. UNLESS that node type does not exist (e.g. plug-in is not available?) In that case you may first need to check also if the plug-in is loaded, etc.

Reply:
i have tried with making the single list with cmds.ls, but it can find all the ornatrix-related nodes instead of being those who connected to the geometry which needs to be published.

Reply:
> Collaborator

The second one works but the first one would error out as the flag only accepts string value.